django_project/clip-and-ship/views.py

```diff
@@ -43,21 +45,74 @@ def clip_layer(request, layername):
     raster_filepath = None
     extention = ''
 
-    file_names = []
-    for layerfile in layer.upload_session.layerfile_set.all():
-        file_names.append(layerfile.file.path)
+    # get file for raster
+    try:
+        if not raster_filepath:
+            file_names = []
+            for layerfile in layer.upload_session.layerfile_set.all():
+                file_names.append(layerfile.file.path)
+
+            for target_file in file_names:
+                if '.tif' in target_file:
+                    raster_filepath = target_file
+                    extention = 'tif'
+                    break
+        bbox_array = bbox_string.split(',')
+        southwest_lat = bbox_array[1]
+        bbox_array[1] = bbox_array[3]
+        bbox_array[3] = southwest_lat
+        bbox_string = ','.join(bbox_array)
+    except AttributeError:
+        # Call wcs command
+        wcs_url = 'http://geoserver:8080/geoserver/wcs?request=getcoverage&' \
```

Comment:
Is geoserver running always on this URL? Or shouldn't we read the value from the settings.py?

Reply:
Yep, there is a geoserver location in setting file. I'll update it.